src/NewPost.js

```diff
@@ -0,0 +1,114 @@
+import React, { Component } from "react";
+import { data } from "./data";
+
+export class NewPost extends Component {
+  state = {
+    id: 0,
+    title: "",
+    author: "",
+    createdOn: "",
+    body: "",
+    tags: ""
+  };
+
+  createDateOnSubmit = () => {
+    const date = new Date();
+    let year = date.getFullYear();
+    let month = date.getMonth();
+    let day = date.getDate();
+    const dateNow = `${year}-${month}-${day}`;
+    return dateNow;
+  };
+
+  createUniqueIdOnSubmit = () => {
+    const id = data.length;
+    return id;
+  };
+
+  convertTagsToArray = tags => {
+    let resultArray = [];
+
+    if (tags != null) {
+      const splitOn = ",";
+
+      resultArray = tags.split(splitOn);
+      resultArray = resultArray.map(function(str) {
+        return str.trim();
+      });
+    }
+
+    return resultArray;
+  };
+
+  handleChange = e => {
+    this.setState({
+      [e.target.name]: e.target.value
+    });
+  };
+
+  handleSubmit = e => {
+    e.preventDefault();
+
+    const resultArray = this.convertTagsToArray(this.state.tags);
+    let blogPost = Object.assign({}, this.state);
```

Comment:
note that you could actually use `const` instead of `let` here and still assign properties like `blogPost.tags`. check out https://mathiasbynens.be/notes/es6-const for a discussion of why this is

Reply:
got it. const is not for 'constant' but rather for 'always referring to same object'